sdk/cosmos/azure_data_cosmos/src/options/mod.rs

```diff
@@ -1,44 +1,50 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-mod cosmos_client_options;
-mod create_container_options;
-mod create_database_options;
-mod delete_container_options;
-mod delete_database_options;
-mod item_options;
-mod query_containers_options;
-mod query_databases_options;
-mod query_options;
-mod read_container_options;
-mod read_database_options;
-
-pub use cosmos_client_options::CosmosClientOptions;
-pub use create_container_options::CreateContainerOptions;
-pub use create_database_options::CreateDatabaseOptions;
-pub use delete_container_options::DeleteContainerOptions;
-pub use delete_database_options::DeleteDatabaseOptions;
-pub use item_options::ItemOptions;
-pub use query_containers_options::QueryContainersOptions;
-pub use query_databases_options::QueryDatabasesOptions;
-pub use query_options::QueryOptions;
-pub use read_container_options::ReadContainerOptions;
-pub use read_database_options::ReadDatabaseOptions;
-
-pub mod builders {
-    //! Builders used to create options types.
-    //!
-    //! You shouldn't need to construct these builders on your own. Instead, use the `builder()` method on the related options type to get an instance of the builder.
-
-    pub use super::cosmos_client_options::CosmosClientOptionsBuilder;
-    pub use super::create_container_options::CreateContainerOptionsBuilder;
-    pub use super::create_database_options::CreateDatabaseOptionsBuilder;
-    pub use super::delete_container_options::DeleteContainerOptionsBuilder;
-    pub use super::delete_database_options::DeleteDatabaseOptionsBuilder;
-    pub use super::item_options::ItemOptionsBuilder;
-    pub use super::query_containers_options::QueryContainersOptionsBuilder;
-    pub use super::query_databases_options::QueryDatabasesOptionsBuilder;
-    pub use super::query_options::QueryOptionsBuilder;
-    pub use super::read_container_options::ReadContainerOptionsBuilder;
-    pub use super::read_database_options::ReadDatabaseOptionsBuilder;
+use azure_core::ClientOptions;
+
+/// Options used when creating a [`CosmosClient`](crate::CosmosClient).
+#[derive(Clone, Debug, Default)]
+pub struct CosmosClientOptions {
+    pub client_options: ClientOptions,
 }
+
+/// Options to be passed to [`DatabaseClient::create_container()`](crate::clients::DatabaseClient::create_container()).
```

Comment:
Method options should have a `method_options: typespec_client_core::http::ClientMethodOptions` member.

Reply:
I'm going to take that in a second PR, because I don't currently _use_ the `Context` that provides, and I want to make sure we do that when I add it. I'll start on that right now though.

Reply:
Actually, I'll add it on to this PR, it won't take much. Going to flip this back to draft until that's done.